scripts/core/editor3/components/toolbar/MultiLineQuoteControls.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import {connect} from 'react-redux';
+import StyleButton from './StyleButton';
+import * as actions from '../../actions';
+import {inlineStyles} from '../../helpers/inlineStyles';
+import {getData, getCell} from '../../helpers/table';
+import {SelectionButtonCustomEditorState} from './SelectionButton';
+import {PopupTypes} from '../../actions';
+import {gettext} from 'core/utils';
+import {IEditorStore} from 'core/editor3/store';
+import {blockStyles} from './BlockStyleButtons';
+import {EditorState} from 'draft-js';
+import {LinkInputMultiLineQuote} from '../links/LinkInput';
+import {LinkToolbarMultiLineQuote} from '../links/LinkToolbar';
+
+interface IProps extends Partial<IEditorStore> {
+    className: string;
+    toggleTableStyle(): void;
+    toggleMultiLineQuoteBlockStyle(style: string): void;
+    setMultiLineQuotePopup(popupType: string, payload: any): void;
+    popup: any;
+}
+
+/**
+ * Holds the logic for multi-line quote toolbar operations.
+ */
+const MultiLineQuoteControlsComponent: React.FunctionComponent<IProps> = ({
+    activeCell,
+    editorState,
+    editorFormat,
+    toggleTableStyle,
+    className,
+    setMultiLineQuotePopup,
+    toggleMultiLineQuoteBlockStyle,
+    popup,
+}) => {
+    const {i, j, key, currentStyle, selection} = activeCell;
+    const contentState = editorState.getCurrentContent();
+    const data = getData(contentState, key);
+    const cellEditorState: EditorState = getCell(data, i, j, currentStyle, selection);
+    const currentInlineStyle = cellEditorState.getCurrentInlineStyle();
+    const blockStyle = cellEditorState
+        .getCurrentContent()
+        .getBlockForKey(cellEditorState.getSelection().getStartKey())
+        .getType();
+
+    return (
+        <div className={'table-controls ' + className}>
+            {
+                editorFormat
+                    .filter((type) => type in blockStyles && type !== 'quote')
```

Comment:
where do we filter out unsupported options like 'media', 'table', 'comments', 'annotations', 'suggestions'?

Reply:
We filter them here. They come from elsewhere. They aren't part of `blockStyles` nor `inlineStyles` so they get filtered out.